### PR TITLE
Feature/pokedex share pokemon urls

### PR DESCRIPTION
**Pull request by Glup3**
# Motivation
You can't share pokemon entries via URL.

# Solution
Make use of query paremeters.

https://user-images.githubusercontent.com/25965213/235289575-a07cbe69-8f13-4163-bb21-53dc8b4e0674.mp4

### CODE REVIEW

File: src/hooks/usePokemonQuery.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { useQuery } from './useQuery';
+import { MAX_CURRENT_POKEMON } from '../components/Pokedex/pokedexConstants';
+import { useHistory, useLocation } from '@docusaurus/router';
+
+const POKEMON_QUERY_PARAMETER_NAME = 'pokemon';
+
+const isValidPokemonDexId = (pokemonDexId) => pokemonDexId >= 0 && pokemonDexId <= MAX_CURRENT_POKEMON;
```
Comment (Denmark14) [inline src/hooks/usePokemonQuery.jsx:8]: Are you using this to search by monsno or ID? 

Pokemon ID is separate from dex ID, monsno would be what you would think of as DexID. 

Either way, 493 is too low of a max and the links will not be what the users expect, so an alternative form of expression is likely required to include monsno and formno.

File: src/components/Pokedex/pokedexConstants.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const MAX_CURRENT_POKEMON = 493;
```
Comment (Denmark14) [inline src/components/Pokedex/pokedexConstants.js:1]: Likely far too low.

File: src/hooks/usePokemonQuery.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+export const usePokemonQuery = () => {
+  const query = useQuery();
+  const history = useHistory();
+  const location = useLocation();
+
+  const parsedPokemonDexId = Math.floor(Number(query.get(POKEMON_QUERY_PARAMETER_NAME))) || 0;
+  const pokemonDexId = isValidPokemonDexId(parsedPokemonDexId) ? parsedPokemonDexId : 0;
+
+  const setPokemonDexId = (pokemonId) => {
+    const params = new URLSearchParams({ [POKEMON_QUERY_PARAMETER_NAME]: pokemonId });
```
Comment (Denmark14) [inline src/hooks/usePokemonQuery.jsx:19]: Searching only by Pokemon ID isn't something we can do, unless we drop form support in links.

**Review by Denmark14 — CHANGES_REQUESTED** [review]
Overall there's some design decisions to be revisited. The code itself is fine though.

File: src/hooks/usePokemonQuery.jsx (hunk 1 above)
Comment (Glup3) [reply src/hooks/usePokemonQuery.jsx:8]: yea - i guess we will have to think of how we wanna do it

File: src/hooks/usePokemonQuery.jsx (hunk 3 above)
Comment (Glup3) [reply src/hooks/usePokemonQuery.jsx:19]: every pokemon form has a unique pokemon id, right? - maybe we should also do it like at smogon & have a dedicated name slug for each form: `venusaurmega`, `venusaur` `venusaurdynamax`, ...

**Comment by Glup3** [comment]
Proof of concept code - once we have more details on how the URL & autocomplete work - we can reuse this code. PR will be closed for now

**Pull request closed without merging**